modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/headless/batch/engine/resource/v1_0/test/ExportTaskResourceTest.java

```diff
@@ -113,6 +126,99 @@ public void testPostExportTask() throws Exception {
 		}
 	}
 
+	@Test
+	public void testPostExportTaskWithFiltering() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.batch.engine.internal." +
+					"BatchEngineExportTaskExecutorImpl",
+				LoggerTestUtil.ERROR)) {
+
+			ObjectDefinition objectDefinition =
+				ObjectDefinitionTestUtil.publishObjectDefinition(
+					Collections.singletonList(
+						ObjectFieldUtil.createObjectField(
+							ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+							ObjectFieldConstants.DB_TYPE_STRING,
+							_OBJECT_FIELD_NAME_TEXT)),
+					ObjectDefinitionConstants.SCOPE_COMPANY,
+					TestPropsValues.getUserId());
+
+			String objectDefinitionRESTContextPath =
+				objectDefinition.getRESTContextPath();
+
+			ObjectEntryTestUtil.addObjectEntry(
+				objectDefinition, _OBJECT_FIELD_NAME_TEXT, "TestObject1");
+			ObjectEntryTestUtil.addObjectEntry(
+				objectDefinition, _OBJECT_FIELD_NAME_TEXT, "TestObject2");
+			ObjectEntryTestUtil.addObjectEntry(
+				objectDefinition, _OBJECT_FIELD_NAME_TEXT, "Object3");
+
+			String queryParametersString = StringBundler.concat(
+				"restrictFields=actions&filter=contains%28",
+				_OBJECT_FIELD_NAME_TEXT, "%2C%20%27Test%27%29");
+
+			JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
+				null,
+				StringBundler.concat(
+					"headless-batch-engine/v1.0/export-task",
+					"/com.liferay.object.rest.dto.v1_0.ObjectEntry/JSON?",
+					"taskItemDelegateName=", objectDefinition.getName(), "&",
+					queryParametersString),
+				Http.Method.POST);
+
+			String actualExecuteStatus = null;
+
+			while (true) {
+				jsonObject = HTTPTestUtil.invokeToJSONObject(
+					null,
+					StringBundler.concat(
+						"headless-batch-engine/v1.0/export-task",
+						"/by-external-reference-code/",
+						jsonObject.getString("externalReferenceCode")),
+					Http.Method.GET);
+
+				actualExecuteStatus = jsonObject.getString("executeStatus");
+
+				if (StringUtil.equals(actualExecuteStatus, "COMPLETED") ||
+					StringUtil.equals(actualExecuteStatus, "FAILED")) {
+
+					break;
+				}
+			}
+
+			InputStream inputStream = HTTPTestUtil.invokeToInputStream(
+				null,
+				StringBundler.concat(
+					"headless-batch-engine/v1.0/export-task",
+					"/by-external-reference-code/",
+					jsonObject.getString("externalReferenceCode"), "/content"),
+				Http.Method.GET);
+
+			ZipInputStream zipInputStream = new ZipInputStream(inputStream);
+
+			zipInputStream.getNextEntry();
+
+			String responseString = StringUtil.read(zipInputStream);
+
+			JSONObject filteredPageJSONObject = HTTPTestUtil.invokeToJSONObject(
+				null,
+				StringBundler.concat(
+					objectDefinitionRESTContextPath, "/?",
+					queryParametersString),
+				Http.Method.GET);
+
+			JSONArray itemsJSONArray = filteredPageJSONObject.getJSONArray(
+				"items");
+
+			JSONAssert.assertEquals(
+				itemsJSONArray.toString(), responseString,
```

Comment:
You delegate the assertion of your test to the response of the GET endpoint. I think you should know which items should be returned, as you have created them in the test. Maybe better to check it manually instead of calling the GET Endpoint